jena-arq/src/test/java/org/apache/jena/sparql/core/assembler/TestDatasetAssembler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jena.sparql.core.assembler;
+
+import static java.nio.file.Files.createTempFile;
+import static org.apache.jena.assembler.JA.data;
+import static org.apache.jena.assembler.Mode.DEFAULT;
+import static org.apache.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.apache.jena.riot.Lang.NQUADS;
+import static org.apache.jena.riot.RDFDataMgr.write;
+import static org.apache.jena.vocabulary.RDF.type;
+
+import java.io.*;
+import java.nio.file.Path;
+
+import org.apache.jena.assembler.JA;
+import org.apache.jena.assembler.exceptions.CannotConstructException;
+import org.apache.jena.query.Dataset;
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.sparql.core.*;
+import org.apache.jena.sparql.sse.SSE;
+import org.apache.jena.sparql.util.IsoMatcher;
+import org.apache.jena.system.Txn;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestDatasetAssembler extends Assert {
```

Comment:
Having one or two tests that works from a Turtle string or would make for clear tests of expected use.

With test for bad cases of mixed usage (general and TIM vocabulary together)  text written test cases will help generating tests.

Reply:
Sure, a couple of Turtle assembler files would make for nice examples. Do we have a standard practice about how to load test files from the classpath during a build?

I don't understand the second sentence. Can you restate that?

Reply:
Generally, tests are loaded from a `testing/` directory, not the classpath.  Some tests need to read real, normal files and also the W3C SPARQL tests run from on-disk files and manifests.  Once some tests do need files, there isn't much value in classpath loading as its extra, not instead of.  Assemblers need to read from files, it being the common usecase.

Reply:
I was lookign for tests that do both `ja:data` and `ja:defaultGraph`, which should be an error. I saw someone asking recently trying with `ja:defaultGraph` and TIM assemble, which just ignores unknown properties.  Easy confusion to make - so deliberately testing making an error seems sensible.